ethpm/backends/ipfs.py

```diff
@@ -56,7 +61,14 @@ def __init__(self) -> None:
 
     def fetch_uri_contents(self, uri: str) -> bytes:
         ipfs_hash = extract_ipfs_path_from_uri(uri)
-        return self.client.cat(ipfs_hash)
+        contents = self.client.cat(ipfs_hash)
+        validation_hash = generate_file_hash(contents)
+        if validation_hash != ipfs_hash:
+            raise ValidationError(
+                "Hashed IPFS contents retrieved from uri: {0} "
```

Comment:
If you're python 3.6+ you can start using f-strings for formatting: https://realpython.com/python-f-strings/